python/lsst/ts/externalscripts/maintel/warmup_hexapod.py

```diff
@@ -213,31 +212,133 @@ async def warm_up(self):
         for step, sleep_time in zip(self.config.step_size, self.config.sleep_time):
             await self.single_loop(step, sleep_time)
 
-    async def move_stepwise(self, start, stop, step, sleep_time):
-        """Moves the hexapod from `start` to `stop` in `steps`.
+    async def move_stepwise(
+        self, start: float, stop: float, step_initial: float, sleep_time: float
+    ):
+        """Moves the hexapod from `start` to `stop` that begins from
+        `step_initial`. The step size is updated based on the result of the
+        movement.
 
         Parameters
         ----------
-        start : float
+        start : `float`
             Initial position.
-        stop : float
+        stop : `float`
             Final position.
-        step : float
-            Step size.
-        sleep_time : float
+        step_initial : `float`
+            Initial step size.
+        sleep_time : `float`
             Time between movements.
         """
-        axes = "xyzuvw"
 
-        for new_pos in np.arange(start, stop, step):
-            pos = await self.hexapod.tel_application.aget()
+        position_current = start
+        position_next = start
+        step_current = step_initial
+        while position_current != stop:
+            # Make sure the next position is not beyond the stop
+            position_next = position_current + step_current
+            if step_initial >= 0:
+                if position_next >= stop:
+                    position_next = stop
+            else:
+                if position_next <= stop:
+                    position_next = stop
+
+            # Do the movement
+            all_positions = await self._get_position()
+            all_positions[self.config.axis] = position_next
+
+            is_successful = await self._move_hexapod(**all_positions)
+
+            # Update the step size based on the result
```

Comment:
I wonder if we should add some verbosity here regarding the changes in the step size. Especially because, if I understand correctly, the step size will be updated on every step. It will either be larger or smaller. But never the same. Is this correct?

Reply:
Do you mean to add more comment to explain the update process of step size? Or you want to me to add `self.log.info()` or `self.log.debug()` to log the new step size?

Reply:
This should be resolved in the commit: [0349e62](https://github.com/lsst-ts/ts_externalscripts/pull/172/commits/0349e6263737f4acad2f602f66a29bfb99c9b8a8). Thanks!